SpiceSharp/Simulations/IntegrationMethods/Spice/Gear/GearStateDerivative.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using SpiceSharp.Simulations;
+
+namespace SpiceSharp.IntegrationMethods
+{
+    public partial class Gear
+    {
+        /// <summary>
+        /// A state that can be derived by the Gear method 
+        /// </summary>
+        protected class GearStateDerivative : StateDerivative, ITruncatable
+        {
+            // Private variables
+            private readonly int _index;
+            private readonly Gear _method;
+            private readonly History<IntegrationState> _states;
+
+            /// <summary>
+            /// Gets or sets the current value
+            /// </summary>
+            public override double Current
+            {
+                get => _states[0].State[_index];
```

Comment:
@svenboulanger Looks like place where null reference could be. I don't know the algorithm but what is _states[0]. It's not clear.  Maybe introducing const would help.

Reply:
`_states[0]` is the current integration state. So this property would set or get the *current* value of the state.
The size of the `History<>` is `maxOrder+2`, and `maxOrder` is minimally `1`, as required by the base class `IntegrationMethod`.